### DISCUSSION TITLE
clear CheckboxList data

### DISCUSSION THREAD

**Discussion by NazarAli** [filamentphp/filament#9013] [General]
i have problem when try clear the CheckboxList data by $set('monthDays',[]);
the all check box unchecked but still value in database

**Comment by joshembling** [comment]
Try using `mutateFormDataBeforeSave` function on your edit page to customise the data 

https://filamentphp.com/docs/3.x/panels/resources/editing-records#customizing-data-before-saving

**Reply by NazarAli** [reply]
the checkboxlist it relationship field 
and i  try print the data in methods mutateFormDataBeforeSave  not found the relationship field

**Reply by joshembling** [reply]
Let's see some code

**Reply by NazarAli** [reply]
![Screenshot 2023-10-11 013346](https://github.com/filamentphp/filament/assets/7056487/961f2fa6-0bb9-46ea-8f89-11f84649e6f0)


i have tow list of checkbox show on and hide another one and clear when show or hide
